internal/store/job.go

```diff
@@ -350,3 +383,10 @@ func createJobListWatch(kubeClient clientset.Interface, ns string) cache.ListerW
 		},
 	}
 }
+
+func failureReason(jc *v1batch.JobCondition, reason string) bool {
```

Comment:
I don't think the job condition is needed here

Reply:
Basically, here we want to check if JobCondition's reason is one of the `jobFailureReasons` - `{"BackoffLimitExceeded", "DeadLineExceeded", "Evicted"}`. 
If it's true then we return the reason otherwise we return an empty string.

Reply:
Indeed, I misread the code. I thought you were only checking if the condition was nil.